src/main/java/de/tum/in/www1/artemis/service/TextBlockService.java

```diff
@@ -20,25 +23,42 @@ public void prepopulateFeedbackBlocks(Result result) throws ClassCastException {
         }
 
         final TextSubmission textSubmission = (TextSubmission) result.getSubmission();
-        final String submissionText = textSubmission.getText();
-        final List<String> blocks = splitSubmissionIntoBlocks(submissionText);
-        final List<Feedback> feedbacks = blocks.stream().map(block -> (new Feedback()).reference(block).credits(0d)).collect(Collectors.toList());
+        final List<TextBlock> blocks = splitSubmissionIntoBlocks(textSubmission);
+        final List<Feedback> feedbacks = blocks.stream().map(block -> (new Feedback()).reference(block.getText()).credits(0d)).collect(toList());
 
         result.getFeedbacks().addAll(feedbacks);
     }
 
-    public List<String> splitSubmissionIntoBlocks(String submission) {
+    @Transactional(readOnly = true)
```

Comment:
JavaDoc missing. I think also more comments in the method would be nice.

Reply:
✅